### ISSUE TITLE
Issue syncing with Nextcloud

### ISSUE THREAD

**Issue by Nokse22** [mrvladus/Errands#167]
Hi, I have a nextcloud account linked to GNOME, I can access files from Nautilus, so it should work.
I select Nextcloud as a sync provider and the password, account name and server url are taken correctly, but when I try to sync or test the connection I get:

```
[INFO] Sync: Initialize sync provider
[INFO] Sync: Initialize 'Nextcloud' sync provider
[DEBUG] Sync: Checking credentials
Nextcloud
[DEBUG] Sync: Checking URL
[DEBUG] Sync: URL is set to https://my_mail%40gmail.com@use08.thegood.cloud/remote.php/dav/
[DEBUG] Sync: Attempting connection
Exception in thread Thread-1 (sync):
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/app/share/errands/errands/lib/sync/sync.py", line 47, in sync
    self.init(self.window)
  File "/app/share/errands/errands/lib/sync/sync.py", line 29, in init
    self.provider = SyncProviderNextcloud(
                    ^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/errands/errands/lib/sync/providers/nextcloud.py", line 8, in __init__
    return super().__init__(name="Nextcloud", *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/errands/errands/lib/sync/providers/caldav.py", line 29, in __init__
    return self._connect()
           ^^^^^^^^^^^^^^^
  File "/app/share/errands/errands/lib/sync/providers/caldav.py", line 61, in _connect
    with DAVClient(
         ^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/caldav/davclient.py", line 385, in __init__
    password = unquote(self.url.password)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/urllib/parse.py", line 671, in unquote
    if '%' not in string:
       ^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
[DEBUG] GOA: Getting data for Nextcloud
```
I have censured my mail like `my_mail%40gmail.com`, I have changed just `my_mail`, the `@` wasn't present
and it's not presented an error with a toast.

Let me know if there is anything I can do to help debug this.

**Comment by Nokse22** [comment]
I managed to get it working by changing the server url to the correct one and the username to my email.
I think you might need to check that the provided url is in a correct format before trying to connect and if not display a toast saying so.
Also, when testing the connection the UI freezes.

**Comment by Nokse22** [comment]
A possible solution is to put the code to connect in a try except

**Comment by mrvladus** [comment]
Fixed in [45.1.7](https://github.com/mrvladus/Errands/releases/tag/45.1.7)
